at_client/lib/src/client/at_client_impl.dart

```diff
@@ -308,8 +310,69 @@ class AtClientImpl implements AtClient {
   @override
   Future<bool> put(AtKey atKey, dynamic value,
       {bool isDedicated = false}) async {
-    // Perform atKey validations.
-    await AtClientValidation.validateAtKey(atKey);
+    AtResponse atResponse = AtResponse();
+    // If the value is neither String nor List<int> throw exception
+    if (value is! String && value is! List<int>) {
+      // TODO: Throw AtValueException
+      throw AtClientException('AT014',
+          'Invalid value type found ${value.runtimeType}. Expected String or List<int>');
+    }
+    if (value is String) {
+      atResponse = await putText(atKey, value);
+    }
+    if (value is List<int>) {
+      atResponse = await putBinary(atKey, value);
+    }
+    return atResponse.response.isNotEmpty;
+  }
+
+  /// put's the text data into the keystore
+  @override
+  Future<AtResponse> putText(AtKey atKey, String value) async {
+    // Set the default metadata if not already set.
+    atKey.metadata ??= Metadata();
+    // Setting metadata.isBinary to false for putText
+    atKey.metadata!.isBinary = false;
+    return await _putInternal(atKey, value);
+  }
+
+  /// put's the binary data(e.g. images, files etc) into the keystore
+  @override
+  Future<AtResponse> putBinary(AtKey atKey, List<int> value) async {
+    // Set the default metadata if not already set.
+    atKey.metadata ??= Metadata();
+    // Setting metadata.isBinary to true for putBinary
+    atKey.metadata!.isBinary = true;
+    // Base2e15.encode method converts the List<int> type to String.
+    return await _putInternal(atKey, Base2e15.encode(value));
+  }
+
+  Future<AtResponse> _putInternal(AtKey atKey, dynamic value) async {
+    // Performs the put request validations.
+    AtClientValidation.validatePutRequest(atKey, value, preference!);
```

Comment:
I was thinking of making this a validator

Reply:
1. Who has the knowledge of throwing the right exceptions for the scenarios we have documented. (atClient or atValidator or Somethingelse)
2. Though encryption and decryption do not have any correlation to the validators, however, they do throw the exceptions. Is there an abstraction we can add around the throwing of exceptions